eoxserver/services/ows/wps/processes/get_statistics.py

```diff
@@ -0,0 +1,146 @@
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies of this Software or works derived from this Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+# -----------------------------------------------------------------------------
+
+import numpy as np
+
+from eoxserver.core import Component
+
+from eoxserver.contrib import gdal
+
+from eoxserver.services.ows.wps.parameters import (
+    LiteralData, ComplexData, FormatJSON, CDObject, BoundingBoxData
+)
+
+from eoxserver.resources.coverages import models
+from eoxserver.backends.access import gdal_open
+from django.db.models import Q, F
+
+from django.contrib.gis.geos import Polygon
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class GetStatisticsProcess(Component):
+    """ GetStatistics defines a WPS process for Raster image Statistics
+        retrieval"""
+
+    identifier = "GetStatistics"
+    title = "Get statistics for a coverage/s that intersects with the input bbox"
+    description = ("provides statistics of all the coverages whithin a provided bounding box. "
+                  " The process is used by the  by the EOxC DEM implementtation")
+    metadata = {}
+    profiles = ['EOxServer:GetStatistics']
+
+    inputs = {
+        "bbox": BoundingBoxData(
+            "bbox",
+            title="bounding box that intersect with the products."
+        ),
+        "collection": LiteralData(
+            "collection",
+            title="The Identifier of the collection of intrest."
+        ),
+    }
+
+    outputs = {
+        "statistic": ComplexData(
+            "statistic",
+            title="output statistics",
+            abstract="coverage/s statistics in json format.",
+            formats=FormatJSON()
+        ),
+    }
+
+    @staticmethod
+    def execute(bbox, collection, **kwarg):
+        """ The main execution function for the process.
+        """
+
+        np_bbox = np.array(bbox)
+        flattened_bbox = np_bbox.flatten()
+        values = flattened_bbox.tolist()
+
+        parsed_bbox = Polygon.from_bbox(values)
+
+        # get the dataset series intersecting with the requested bbox
+        collection = models.Collection.objects.get(identifier=collection)
+
+        coverages = models.Coverage.objects.filter(
+            Q(collections=collection.id) &
+            Q(footprint__intersects=parsed_bbox)
+            | Q(footprint__isnull=True, parent_product__footprint__intersects=parsed_bbox))
```

Comment:
There is a subtle bug in this: when you use brackets to group these queries they actually look like this:

```python
(
  Q(collections=collection.id) &
  Q(footprint__intersects=parsed_bbox)
)
| Q(footprint__isnull=True, parent_product__footprint__intersects=parsed_bbox))
```

... which is not what we want. We want the collection check to be always true, but the footprint can either be on the coverage or the product. So we have to explicitly state that using brackets:

```python

Q(collections=collection.id) &
(
  Q(footprint__intersects=parsed_bbox)
  | Q(footprint__isnull=True, parent_product__footprint__intersects=parsed_bbox))
)
```